program/src/processor/create_tokenized_attestation.rs

```diff
@@ -0,0 +1,316 @@
+use core::marker::PhantomData;
+
+use pinocchio::{
+    account_info::AccountInfo,
+    instruction::{Seed, Signer},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvars::{rent::Rent, Sysvar},
+    ProgramResult,
+};
+use pinocchio_associated_token_account::instructions::Create;
+use pinocchio_log::log;
+use pinocchio_token::{
+    extensions::{
+        group_member_pointer::Initialize as InitializeGroupMemberPointer,
+        metadata::{Field, InitializeTokenMetadata, UpdateField},
+        metadata_pointer::Initialize as InitializeMetadataPointer,
+        non_transferable::InitializeNonTransferableMint,
+        token_group::InitializeMember,
+    },
+    instructions::{InitializeMint2, MintToChecked, TokenProgramVariant},
+    TOKEN_2022_PROGRAM_ID,
+};
+use solana_program::pubkey::Pubkey as SolanaPubkey;
+
+use crate::{
+    constants::{ATTESTATION_MINT_SEED, SAS_SEED, SCHEMA_MINT_SEED},
+    error::AttestationServiceError,
+    processor::process_create_attestation,
+};
+
+use super::{
+    create_pda_account, verify_ata_program, verify_system_account, verify_token22_program,
+};
+
+#[inline(always)]
+pub fn process_create_tokenized_attestation(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    // Create Attestation first
+    process_create_attestation(program_id, &accounts[0..6], instruction_data)?;
+
+    let [payer_info, _authorized_signer, _credential_info, schema_info, attestation_info, system_program, schema_mint_info, attestation_mint_info, sas_pda_info, recipient_token_account_info, recipient_info, token_program, ata_program] =
+        accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    // Validate: should be owned by system account, empty, and writable
+    verify_system_account(recipient_token_account_info, true)?;
+
+    // Verify token programs.
+    verify_token22_program(token_program)?;
+    verify_ata_program(ata_program)?;
+
+    if recipient_info.is_writable() {
+        log!("Recipient should not be writable");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    // Validate that mint matches expected PDA
+    let (attestation_mint_pda, attestation_mint_bump) = SolanaPubkey::find_program_address(
+        &[ATTESTATION_MINT_SEED, attestation_info.key()],
+        &SolanaPubkey::from(*program_id),
+    );
+    if attestation_mint_info
+        .key()
+        .ne(&attestation_mint_pda.to_bytes())
+    {
+        return Err(AttestationServiceError::InvalidMint.into());
+    }
+    let (schema_mint_pda, _) = SolanaPubkey::find_program_address(
+        &[SCHEMA_MINT_SEED, schema_info.key()],
+        &SolanaPubkey::from(*program_id),
+    );
+
+    if schema_mint_info.key().ne(&schema_mint_pda.to_bytes()) {
+        return Err(AttestationServiceError::InvalidMint.into());
+    }
```

Comment:
Should we verify that a Schema Mint actually exists? Right now we check to make sure the PDA is correct, but I don't think we're checking if it's initialized.

Reply:
I don't think that's needed. Schema mint is a writable account in the token extensions so those CPIs would fail if it doesn't exist.